datadog_lambda/tracing.py

```diff
@@ -660,18 +623,32 @@ def is_lambda_context():
 
 def set_dd_trace_py_root(trace_context_source, merge_xray_traces):
     if trace_context_source == TraceContextSource.EVENT or merge_xray_traces:
-        context = dict(dd_trace_context)
+        global dd_trace_context
```

Comment:
Changing this to a global alters how this function works. If merge xray traces is True, then you're mutating the global trace context object, which was not done before.